src/cpu.rs

```diff
@@ -65,9 +65,8 @@ impl CPU {
     }
 
     pub fn reset(&mut self) {
-        self.ip = 0;
-        self.instruction_count = 0;
-        // XXX clear memory
+        let cpu = CPU::new();
+        *self = cpu;
```

Comment:
Oh, this is nice!

Reply:
Well, it's clean but not very optimal (unless the compiler can do some magic in the background??)

Reply:
No idea. I didn't even realize we could do this :)